core/src/main/java/jenkins/util/SystemProperties.java

```diff
@@ -308,6 +312,91 @@ public static Boolean optBoolean(String name) {
+     * @param unit the duration unit to use if the value doesn't specify one (defaults to `ms`)
+     * @return the property value as a duration.
+     * @since TODO
+     */
+    @CheckForNull
+    public static Duration getDuration(@NonNull String name, @CheckForNull ChronoUnit unit) {
+        return getDuration(name, unit, null);
+    }
+
+    /**
+     * Determines the duration value of the system property with the specified name, or a default value.
+     * @param name property name.
+     * @param defaultValue a default value
+     * @return the property value as a duration.
+     * @since TODO
+     */
+    @Nullable
+    public static Duration getDuration(@NonNull String name, @CheckForNull Duration defaultValue) {
+        return getDuration(name, defaultValue, Level.CONFIG);
+    }
+
+    /**
+     * Determines the duration value of the system property with the specified name, or a default value.
+     *
+     * @param name         property name.
+     * @param unit         the duration unit to use if the value doesn't specify one (defaults to `ms`)
+     * @param defaultValue a default value
+     * @return the property value as a duration.
+     * @since TODO
+     */
+    @Nullable
+    public static Duration getDuration(@NonNull String name, @CheckForNull ChronoUnit unit, @CheckForNull Duration defaultValue) {
+        return getDuration(name, unit, defaultValue, Level.CONFIG);
+    }
+
+    /**
+     * Determines the duration value of the system property with the specified name, or a default value.
+     * @param name property name.
+     * @param defaultValue a default value
+     * @param logLevel the level of the log if the provided system property name cannot be parsed into Duration.
+     * @return the property value as a duration.
+     * @since TODO
+     */
+    @Nullable
+    public static Duration getDuration(@NonNull String name, @CheckForNull Duration defaultValue, @NonNull Level logLevel) {
+        return getDuration(name, null, defaultValue, logLevel);
+    }
+
+    /**
+     * Determines the duration value of the system property with the specified name, or a default value.
+     *
+     * @param name         property name.
+     * @param unit         the duration unit to use if the value doesn't specify one (defaults to `ms`)
+     * @param defaultValue a default value
+     * @param logLevel     the level of the log if the provided system property name cannot be parsed into Duration.
+     * @return the property value as a duration.
+     * @since TODO
+     */
+    @Nullable
+    public static Duration getDuration(@NonNull String name, @CheckForNull ChronoUnit unit, @CheckForNull Duration defaultValue, @NonNull Level logLevel) {
+        String v = getString(name);
+        if (v != null) {
+            try {
+                return DurationStyle.detectAndParse(v, unit);
+            } catch (Exception e) {
+                LOGGER.log(logLevel, e, () -> "Property. Value is not a duration: " + name + " => " + v);
```

Comment:
this may be better:
```suggestion
                LOGGER.log(logLevel, e, () -> "Failed to convert %s to a valid duration for property %s, falling back to default %s",  v, name, defaultValue);
```